source/solvers/heat_transfer_assemblers.cc

```diff
@@ -164,8 +164,10 @@ HeatTransferAssemblerCore<dim>::assemble_rhs(
                       9 * std::pow(4 * alpha / (h * h), 2));
 
       // Calculate the strong residual for GLS stabilization
-      strong_residual_vec[q] += rho_cp * velocity * temperature_gradient -
-                                thermal_conductivity[q] * temperature_laplacian;
+      strong_residual_vec[q] +=
+        rho_cp * velocity * temperature_gradient -
+        thermal_conductivity[q] * temperature_laplacian +
+        scratch_data.source[q];
```

Comment:
I think there is a sign error here. It should be - and not +.
The equation is rho_cp * u * grad(T) - k * laplacian T - source = 0
This is also what is coded in the RHS
k * (grad T) * grad(test) + rho_cp*u*grad(T) - source 

Only the laplacian changes sign due to integration by part.

Reply:
It is corrected. The errors also decreased in the failed application test. :)